src/v/cloud_storage/remote.cc

```diff
@@ -53,13 +53,45 @@ remote::remote(
         std::visit([](auto&& cfg) { return cfg.disable_metrics; }, conf))),
       remote_metrics_disabled(static_cast<bool>(std::visit(
         [](auto&& cfg) { return cfg.disable_public_metrics; }, conf))),
-      *_materialized) {
-    // If the credentials source is from config file, bypass the background op
-    // to refresh credentials periodically, and load pool with static
+      *_materialized)
+  , _azure_shared_key_binding(
+      config::shard_local_cfg().cloud_storage_azure_shared_key.bind()) {
```

Comment:
This is fine to set even in environments which do not use azure, right? The property is still there, it just isn't ever set or changed?

Reply:
> The property is still there, it just isn't ever set or changed?

Yes, that's right.